### ISSUE TITLE
Libluabind does not work

### ISSUE THREAD

**Issue by pirogovsky85** [Project-OSRM/osrm-backend#2763]
Tried compile on Ubuntu 14.04.5 LTS.
Tried fixes from here #1925 
Installed [https://github.com/mapbox/luabind](https://github.com/mapbox/luabind) :
```
mkdir build
pushd build
cmake .. -DCMAKE_BUILD_TYPE=Release
cmake --build .
sudo make install
```

Check lua version linking:
```
 ldd /usr/local/lib/libluabind.so
        linux-vdso.so.1 =>  (0x00007fffcf851000)
        liblua5.1.so.0 => /usr/lib/x86_64-linux-gnu/liblua5.1.so.0 (0x00007f68de272000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f68ddf6e000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f68ddd57000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f68dd992000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f68dd68c000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f68dd487000)
        /lib64/ld-linux-x86-64.so.2 (0x00005635507ca000)
```

After tried compile:
```
sudo cmake .. -DLUABIND_LIBRARY=/usr/local/lib/libluabind.a -DLUABIND_INCLUDE_DIR=/usr/local/include/ -DCMAKE_BUILD_TYPE=Release
-- Building on a 64 bit system
-- Configuring OSRM in release mode
-- LTO working
-- Disabling LTO on GCC < 4.9.0 since it is broken, see: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=57038
-- Setting linker optimizations
-- Boost version: 1.54.0
-- Found the following Boost libraries:
--   date_time
--   filesystem
--   iostreams
--   program_options
--   regex
--   system
--   thread
-- Found Intel TBB
-- TBB interface version: 7000
-- Looking for Luabind...
-- Found Luabind: /usr/local/lib/libluabind.a
-- Looking for LuaJIT 5.2
-- Found LuaJIT: /usr/lib/x86_64-linux-gnu/libluajit-5.1.so
-- Performing Test LUABIND_WORKS
-- Performing Test LUABIND_WORKS - Failed
-- Luabind/Lua5.2 not feasible, falling back to Lua 5.1.
-- Looking for LuaJIT 5.1
-- Found LuaJIT: /usr/lib/x86_64-linux-gnu/libluajit-5.1.so
-- My message Boost /usr/include
-- My message LUABIND INCLUDE DIR /usr/local/include
-- My message LUA INCLUDE DIR /usr/include/lua5.1
-- My message LUABIND LIBRARY /usr/local/lib/libluabind.a
-- My message LUA LIBRARY /usr/lib/x86_64-linux-gnu/liblua5.1.so
-- Performing Test LUABIND51_WORKS
-- Performing Test LUABIND51_WORKS - Failed
-- Configuring incomplete, errors occurred!
CMake Error at cmake/check_luabind.cmake:45 (message):
  Luabind does not work with Lua 5.1 at
  /usr/lib/x86_64-linux-gnu/liblua5.1.so, no working Luabind found
Call Stack (most recent call first):
  CMakeLists.txt:284 (include)
See also "/mnt/data/osrm/osrm-backend/build/CMakeFiles/CMakeOutput.log".
See also "/mnt/data/osrm/osrm-backend/build/CMakeFiles/CMakeError.log".
```

**Comment by rkcpi** [comment]
This might also be fixed with #2805. Ubuntu 14.04 doesn't support c++14.

**Comment by springmeyer** [comment]
Yes, thanks @rkcpi - this should be fixed by #2805. Closing therefore. Please re-open new tickets if there are still problems.
